ui/vite.config.js

```diff
@@ -4,10 +4,44 @@ import { defineConfig } from "vite";
 import vue from "@vitejs/plugin-vue";
 import monacoEditorPlugin from "vite-plugin-monaco-editor";
 
+const streamsyncComponentsDocs = () => {
+	let config;
+
+	return {
+		name: "vue-docs-ignore",
+		configResolved(resolvedConfig) {
+			config = resolvedConfig;
+		},
+		transform(code, id) {
+			if (/vue&type=docs/.test(id)) {
+				const docs = code
+					.replaceAll(/'/g, "\\'")
+					.replaceAll(/\n/g, "\\n")
+					.replaceAll(/\r/g, "\\r")
+					.trim()
+					.replace(/^(\\n|\\t|[ \s])*/, "")
+					.replace(/(\\n|\\t|[ \s])*$/, "");
+				return `export default Comp => {
+					if(!Comp.streamsync) return;
+					Comp.streamsync.docs = '${docs}';
+				}`;
+			}
+			if (/\.vue$/.test(id)) {
+				if (config.includeStreamsyncComponentPath === false) return;
+				const fileRef = id.replace(__dirname, "");
+				return `${code}
+					if(_sfc_main.streamsync) _sfc_main.streamsync.fileRef = '${fileRef}';
+				`;
+			}
+		},
+	};
+};
+
 // https://vitejs.dev/config/
 export default defineConfig({
 	base: "./",
-	plugins: [vue(), monacoEditorPlugin({})],
+	plugins: [vue(), monacoEditorPlugin({}), streamsyncComponentsDocs()],
```

Comment:
Does this work for custom components? The plugin probably needs to be added there too.

Reply:
`vite.config.custom.js` that is.